stock_location_orderpoint/models/stock_location_orderpoint.py

```diff
@@ -0,0 +1,423 @@
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from copy import copy
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+from odoo.tools import float_compare, split_every
+
+from odoo.addons.stock.models.stock_move import PROCUREMENT_PRIORITIES
+
+
+class StockLocationOrderpoint(models.Model):
+    _name = "stock.location.orderpoint"
+    _description = "Stock location orderpoint"
+    _order = "priority desc, sequence"
+
+    name = fields.Char(
+        "Name",
+        copy=False,
+        required=True,
+        readonly=True,
+        default=lambda self: self.env["ir.sequence"].next_by_code(
+            "stock.location.orderpoint"
+        ),
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        "Company",
+        required=True,
+        default=lambda self: self.env.company,
+    )
+    location_id = fields.Many2one(
+        "stock.location",
+        "Location",
+        ondelete="cascade",
+        required=True,
+        check_company=True,
+        domain='[("company_id", "=", company_id)]',
+    )
+    trigger = fields.Selection(
+        [("auto", "Auto/realtime"), ("manual", "Manual"), ("cron", "Scheduled")],
+        "Trigger",
+        default="auto",
+        required=True,
+        help="Auto/realtime orderpoints are triggered on new moves\n"
+        "Manual orderpoints are triggered via the orderpoints' view\n"
+        "Scheduled orderpoints are triggered via scheduled actions per location",
+    )
+    replenish_method = fields.Selection(
+        [("fill_up", "Fill up")],
+        default="fill_up",
+        required=True,
+        help="Defines how the qty to replenish gets computed\n"
+        "Fill up = The replenishment will be triggered when a move is waiting availability "
+        "and forecast quantity is negative at the location (i.e. min=0). "
+        "The replenished quantity will bring back the forecast quantity to 0 (i.e. max=0) "
+        "but will be limited to what is available at the source location "
+        "to plan only reservable replenishment moves",
+    )
+    sequence = fields.Integer("Sequence", default=10)
+    route_id = fields.Many2one(
+        "stock.location.route",
+        string="Preferred Route",
+        domain="[('rule_ids.location_id', 'in', [location_id])]",
+    )
+    group_id = fields.Many2one(
+        "procurement.group",
+        "Procurement Group",
+        copy=False,
+        help="Moves created through this orderpoint "
+        "will be put in this procurement group. "
+        "If none is given, the moves generated by stock rules "
+        "will be grouped into one big picking.",
+    )
+    location_src_id = fields.Many2one(
```

Comment:
@mt-software-de @jbaudoux If I understand correctly, this is always computed from the route. Could you add some documentation about how to configure orderpoints in USAGE.rst ? Thanks

Shouldn't the _get_location_src_id deserves a separate module for that purpose ?

Reply:
Yes it is computed from the location_id and the route_id
I can move this to the stock_helper module